src/js/select.js

```diff
@@ -188,28 +206,59 @@ function Menu(wrapperEl, selectEl) {
 Menu.prototype._createMenuEl = function(wrapperEl, selectEl) {
   var menuEl = doc.createElement('div'),
       optionEls = selectEl.children,
-      numOptions = optionEls.length,
+      numOptions = calculateOptions(optionEls),
       selectedPos = 0,
       optionEl,
       itemEl,
-      i;
+      i,
+      j,
+      pos = 0;
 
   menuEl.className = menuClass;
 
   // add options
-  for (i=0; i < numOptions; i++) {
+  for (i=0; i < optionEls.length; i++) {
```

Comment:
Yeah, that's probably true. In any case, caching the array length will save a property lookup on each iteration. Here's an SO discussion about loops in JavaScript that includes suggestions about caching the array length:
http://stackoverflow.com/questions/3010840/loop-through-an-array-in-javascript